Signal/src/ViewControllers/ConversationView/ConversationViewItem.h

```diff
@@ -121,8 +121,14 @@ NSString *NSStringForOWSMessageCellType(OWSMessageCellType cellType);
 
 - (BOOL)canSaveMedia;
 
+// For view items that correspond to interactions, this is the interaction's unique id.
+// For other view views (like the typing indicator), this is a unique, stable string.
+- (NSString *)itemId;
```

Comment:
I've introduce the concept of an "item id".  In the past, all conversation items corresponded to a `TSInteraction` persisted in the DB.  Soon, we'll have "transient" conversation items (like typing indicators, unread indicators, date breaks, etc.).